test/integration/hds_integration_test.cc

```diff
@@ -75,56 +67,145 @@ class HdsIntegrationTest : public HttpIntegrationTest,
   }
 
   void cleanupHdsConnection() {
-    if (fake_hds_connection_ != nullptr) {
-      fake_hds_connection_->close();
-      fake_hds_connection_->waitForDisconnect();
+    if (hds_fake_connection_ != nullptr) {
+      hds_fake_connection_->close();
+      hds_fake_connection_->waitForDisconnect();
     }
   }
 
-  static constexpr uint32_t upstream_endpoints_ = 5;
+  envoy::service::discovery::v2::HealthCheckSpecifier makeHealthCheckSpecifier() {
+    // HealthCheckSpecifier
+    envoy::service::discovery::v2::HealthCheckSpecifier server_health_check_specifier;
+    server_health_check_specifier.mutable_interval()->set_nanos(500000000); // 500ms
+    auto* endpoint_address = server_health_check_specifier.add_health_check()
+                                 ->add_endpoints()
+                                 ->add_endpoints()
+                                 ->mutable_address()
+                                 ->mutable_socket_address();
+    endpoint_address->set_address(Network::Test::getLoopbackAddressString(version_));
+    endpoint_address->set_port_value(host_upstream_->localAddress()->ip()->port());
+
+    auto* anna = server_health_check_specifier.mutable_health_check(0)->mutable_endpoints(0);
```

Comment:
You can just latch above `auto* health_check = server_health_check_specifier.add_health_check();`, and use it for both the IP address and the locality population here.

Reply:
Done.